store/store.go

```diff
@@ -408,9 +408,12 @@ func (s *Store[H]) flushLoop() {
 			log.Errorw("writing header batch", "try", i+1, "from", from, "to", to, "err", err)
 			s.metrics.flush(ctx, time.Since(startTime), s.pending.Len(), true)
 
-			const maxRetrySleep = time.Second
-			sleep := min(10*time.Duration(i+1)*time.Millisecond, maxRetrySleep)
-			time.Sleep(sleep)
+			const maxRetrySleep = 100 * time.Millisecond
+			sleepDur := min(10*time.Duration(i+1)*time.Millisecond, maxRetrySleep)
+
+			if err := sleep(ctx, sleepDur); err != nil {
```

Comment:
Isn't that the case that this context is never canceled in practice?

Reply:
Step 1: make to respect context.
Step 2: change context to something real.